### DISCUSSION TITLE
Switch interchangeably between sqlite and postgres

### DISCUSSION THREAD

**Discussion by arthurfiorette** [mikro-orm/mikro-orm#3934] [Q&A] [answered]
Our product needs to have a local instance running on a sqlite db and a cloud version running on a postgresql one.

Does mikro-orm have any kind of support for this situation? I haven't found any mentions about it in the documentation.

**Comment by B4nan** [comment, marked as answer]
As long as you wont be using any postgres specific features, it is just about providing different driver in your config.

Keep in mind that each driver is different, and sqlite specifically has a lot of limitations when it comes to altering of existing schema.

**Comment by arthurfiorette** [comment]
And what about migrations, they'd only apply to what my CLI is connected to. How should I manage running mingrations (and any other CLI related code) to run on my locally SQLite and changing to run it 
on a PostgreSQL.

**Reply by B4nan** [reply]
I'd say you should treat that as two separate apps, two configs, two databases, two migration folders. Sharing migrations between different drivers will almost certainly break.

**Comment by arthurfiorette** [comment]
Yeah, even that both of them works under the SQL directives, it is such a bad idea to try to work with both of them with a single codebase. I'll try to convince my managers that it is going to be a pain in the ass and somehow ship a postgres instance alongside our client.

**Reply by B4nan** [reply]
The application code, as well as entity definition, can be mostly the same (again, if you dont care about postgres specific features, it can be almost the same - e.g. you can use runtime values for defaults instead of dealing with sql dialect differences), but you still need to treat that as two separate apps in terms of migrations.
